compdb/compdb.cc

```diff
@@ -0,0 +1,71 @@
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <fstream>
+#include <iomanip>
+
+#include "nlohmann/json.hpp"
+#include "subprocess.hpp"
+
+using namespace nlohmann;
+typedef std::string string;
+
+string replace(string in, const string& before, const string& after) {
+  const auto idx = in.find(before);
+  if (idx == string::npos) {
+    return in;
+  }
+  return in.replace(idx, before.size(), after);
+}
+
+int main(int argc, const char* argv[]) {
+  // Obviously std::filesystem is quite a bit nicer but it's _relatively_ new (e.g. on the
+  // box I'm writing this, it doesn't seem to be available, even with --std=c++17).
+  // TODO(peterebden): This is not quite right, it always returns the current path, but we
+  //                   really want the repo root. Maybe we should be able to ask plz for it?
```

Comment:
That actually might be something worth implementing. I remember having to implement the walk-up-the-tree-until-plzconfig-is-found a couple of times already.

Reply:
Done this on master now and rewrote this to use it - it's quite a bit nicer than fiddling about with `getcwd` and buffer sizing.